pykafka/balancedconsumer.py

```diff
@@ -416,6 +415,17 @@ def _get_participants(self):
         participants = sorted(participants)
         return participants
 
+    def _build_watch_callback(self, fn, proxy):
+        """Return a function that's safe to use as a ChildrenWatch callback"""
+        def _callback(children):
+            # discover whether the referenced object still exists
+            try:
+                proxy.__repr__()
+            except ReferenceError:
+                return
```

Comment:
A `return False` here will prevent `ChildrenWatch` from calling again after this.

It might also be good to reference the github issue this relates to in the docstring, because it may not be obvious why we did this later. It's a neat solution though!